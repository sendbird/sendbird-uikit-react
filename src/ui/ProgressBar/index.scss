@import '../../styles/variables';

.sendbird-progress-bar, .sendbird-progress-bar__fill {
  display: inline-flex;
  width: 100%;
  height: 100%;
}

.sendbird-progress-bar {
  justify-content: flex-start;
}
.sendbird-progress-bar__fill {
  @include themed() {
    background-color: t(on-bg-3);
  }
}

// colors
.progress-bar-color--primary {
  @include themed() {
    background-color: t(primary-3);
  }
}
.sendbird-progress-bar.progress-bar--disabled, .progress-bar-color--gray {
  @include themed() {
    background-color: t(bg-1);
  }
}
.sendbird-progress-bar.progress-bar--disabled .sendbird-progress-bar__fill {
  background-color: transparent;
}
