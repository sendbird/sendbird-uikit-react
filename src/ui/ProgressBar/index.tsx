import React, { useMemo } from 'react';
import './index.scss';

export enum ProgressBarColorTypes {
  PRIMARY = 'progress-bar-color--primary',
  GRAY = 'progress-bar-color--gray',
}

export interface ProgressBarProps {
  className?: string;
  disabled?: boolean;
  maxSize: number;
  currentSize: number;
  colorType?: ProgressBarColorTypes;
}

export const ProgressBar = ({
  className ='',
  disabled = false,
  maxSize,
  currentSize,
  colorType = ProgressBarColorTypes.PRIMARY,
}: ProgressBarProps): React.ReactElement => {
  const width = useMemo(() => {
    return `${currentSize / maxSize * 100}%`;
  }, [currentSize, maxSize]);
  return (
    <div className={`sendbird-progress-bar ${className} ${colorType} ${disabled ? 'progress-bar--disabled' : ''}`}>
      <div className="sendbird-progress-bar__fill" style={{ width }} />
    </div>
  );
};

export default ProgressBar;
