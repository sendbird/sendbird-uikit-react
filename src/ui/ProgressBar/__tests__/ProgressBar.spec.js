import React from 'react';
import { render } from '@testing-library/react';

import ProgressBar from "../index";

describe('ui/ProgressBar', () => {
  it('should should contain className', function () {
    const text = "example-text";
    const { container } = render(<ProgressBar className={text} />);
    expect(
      container.getElementsByClassName('sendbird-progress-bar')[0].className
    ).toContain('sendbird-progress-bar');
  });

  it('should do a snapshot test of the PlaceHolder DOM', function () {
    const text = "example-text";
    const { asFragment } = render(<ProgressBar className={text} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
