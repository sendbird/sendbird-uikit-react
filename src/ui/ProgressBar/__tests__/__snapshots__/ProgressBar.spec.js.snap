// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/ProgressBar should do a snapshot test of the PlaceHolder DOM 1`] = `
<DocumentFragment>
  <div
    class="sendbird-progress-bar example-text progress-bar-color--primary"
  >
    <div
      class="sendbird-progress-bar__fill"
    />
  </div>
</DocumentFragment>
`;
